git_repo/repo.py

```diff
@@ -209,9 +209,11 @@ def set_repo_slug(self, repo_slug, auto=False):
             self.repo_name = None
         elif '/' in self.repo_slug:
             # in case a full URL is given as parameter, just extract the slug part.
-            #self.namespace, self.repo_name, *overflow = self.repo_slug.split('/')
             *namespace, self.repo_name = self.repo_slug.split('/')
             self.namespace = '/'.join(namespace)
+            if len(namespace) > 1 and not self.service._supports_nested_namespaces:
```

Comment:
here you could do something like `if len(namespace) >= self.service.max_nested_namespaces`